python_modules/dagster-graphql/dagster_graphql/schema/external.py

```diff
@@ -184,18 +185,43 @@ def resolve_location(self, _graphene_info):
         return GrapheneRepositoryLocation(self._repository_location)
 
     def resolve_schedules(self, graphene_info):
-
         schedules = self._repository.get_external_schedules()
+        schedule_states_by_name = {
+            state.name: state
+            for state in graphene_info.context.instance.all_stored_job_state(
+                repository_origin_id=self._repository.get_external_origin_id(),
+                job_type=InstigatorType.SCHEDULE,
+            )
```

Comment:
depending on how the frontend queries are structured this'll be O(num repositories) now, which is def an improvement